juju/juju
# Cmr find list cleanup

## Description of change

Please look at the just second commit as this PR builds on https://github.com/juju/juju/pull/7124
https://github.com/juju/juju/pull/7127/commits/2c965bd3ae11b69f832b03444e5ae8f84ceae022

There's several pieces of cleanup:
- separate CMR list and find APIs
- list and find return different data
-- list includes charm and connection info
- new state support for querying the number of CMR connections
- fix for NPE in firewaller
- log stacktrace when a worker panics for easier diagnosis
- CMR CLI help cleanup

## QA steps

bootstrap and deploy a CMR scenario
run the various CMR commands - list, find, show
check the output
